src/cleanfigure.m

```diff
@@ -251,78 +250,161 @@ function pruneOutsideBox(meta, handle)
+      isInfinite_first  = any(~isfinite(newData_first),  2);
+      isInfinite_second = any(~isfinite(newData_second), 2);
+
+      newData_first(isInfinite_first,  :) = NaN(sum(isInfinite_first),  2);
+      newData_second(isInfinite_second,:) = NaN(sum(isInfinite_second), 2);
+
+      % If a point is part of two segments, that cross the border, we need to
+      % insert a NaN to prevent an additional line segment
+      [trash, trash, id_conflict] = intersect(id_first (~isInfinite_first), ...
+                                              id_second(~isInfinite_second));
+
+      % Cut the data into length(id_replace)+1 segments.
+      % Calculate the length of the segments
+      length_segments = [id_replace(1);
+                         diff(id_replace);
+                         size(data, 1)-id_replace(end)];
+
+      % Create an empty cell array for inserting NaNs and fill it at the
+      % conflict sites
+      dataInsert_NaN              = cell(length(length_segments),1);
+      dataInsert_NaN(id_conflict) = mat2cell(NaN(length(id_conflict), 2),...
+                                             ones(size(id_conflict)), 2);
+
+      % Create a cell array for the moved points
+      dataInsert_first  = mat2cell(newData_first,  ones(size(id_first)),  2);
+      dataInsert_second = mat2cell(newData_second, ones(size(id_second)), 2);
+
+      % Add an empty cell at the end of the last segment
+      dataInsert_first  = [dataInsert_first;  cell(1)];
+      dataInsert_second = [dataInsert_second; cell(1)];
+
+      % If the first or the last point would have been replaced add an empty
+      % cell at the beginning/end
+      if(id_replace(end) == size(data, 1))
+        dataInsert_first  = [dataInsert_first; cell(1)];
+      end
+      if(id_replace(1) == 1)
+        dataInsert_second = [cell(1); dataInsert_second];
+      end
+
+      % Put the cells together, right points first, then the possible NaN
+      % and then the left points      
+      dataInsert = cellfun(@(x,y,z) [x; y; z],...
+                            dataInsert_second,...
+                            dataInsert_NaN,...
+                            dataInsert_first,...
+                            'UniformOutput',false);
+  end
+
+  % Insert the data
+  insertData(handle, id_replace, dataInsert);
   
-  % Check whether lambda is in bound
-  out = 0.0 < lambda(:, 1) & lambda(:, 1) < 1.0 &...
-        0.0 < lambda(:, 2) & lambda(:, 2) < 1.0;
+  % Get the indices of the to be removed points respecting the now inserted
+  % data points
+  id_remove = id_replace + cumsum(cellfun(@(x) size(x,1), [cell(1);dataInsert(1:end-2)]));
+  
+  % Remove the data point to be replaced. This has to happen fist, as it is
+  % not clear, hom many points are to be inserted
+  % Remove the data outside the box
+  removeData(meta, handle, id_remove);
+
+  % Remove possible NaN duplications
+  removeNaNs(meta, handle);
+  return;
```

Comment:
Yeah, i saw it in some function and just plastered it everywhere. Removed locally.